Update minimal_access.sls

puneetk opened the pull request:
Remove for loop in favor of names

puneetk committed:
Update minimal_access.sls

Remove for loop in favor of names

nmadhok commented:
I'm not sure if this is the right way to go. Thoughts?

puneetk commented:
does it not work or a stylistic issue. 

## 

Puneet

> On Apr 4, 2015, at 13:37, Nitin Madhok notifications@github.com wrote:
> 
> I'm not sure if this is the right way to go. Thoughts?
> 
> —
> Reply to this email directly or view it on GitHub.

nmadhok commented on os-hardening/minimal_access.sls:
```diff
@@ -1,7 +1,7 @@
-{{dir}}:
+hardening-read-only-folders:
   file.directory:
+    - names: {{ hardening.read_only_folders }}
```
`names` does not exist as an argument to `file.directory`: http://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.directory

puneetk replied:
hmm my bad, my assumption was names is global to all state functions 

## 

Puneet

> On Apr 4, 2015, at 16:45, Nitin Madhok notifications@github.com wrote:
> 
> In os-hardening/minimal_access.sls:
> 
> >    file.directory:
> > -    - names: {{ hardening.read_only_folders }}
> >   names does not exist as an argument to file.directory: http://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.directory
> 
> —
> Reply to this email directly or view it on GitHub.

puneetk commented:
@nmadhok 
It works fine in 2014.7+ versions 

```
# -*- coding: utf-8 -*-
# vim: ft=sls
{% set read_only_folders = ['/usr/local/sbin','/usr/local/bin','/usr/sbin','/usr/bin','/sbin','/bin'] %}
hardening-read-only-folders:
  file.directory:
    - names: {{ read_only_folders }}
    - user: root
    - group: root
    - follow_symlinks: True
    - dir_mode: 755
    - file_mode: 755
    - recurse:
      - user
      - group
      - mode
```

which works like so, so looks like names is valid construct to use in file, let me know. 

```
root@salt:/srv/salt# salt 'onion' state.sls test
onion:
  Name: /bin - Function: file.directory - Result: Changed
  Name: /sbin - Function: file.directory - Result: Changed
  Name: /usr/bin - Function: file.directory - Result: Changed
  Name: /usr/sbin - Function: file.directory - Result: Changed

Summary
------------
Succeeded: 4 (changed=4)
Failed:    0
------------
```

nmadhok commented:
My bad. I didn't think about the recent versions. I was thinking more about supporting users using older versions. Never mind. Thanks for the change!

puneetk commented:
should we add a disclaimer for supported versions ? I can write one up 

## 

Puneet

> On Apr 4, 2015, at 21:29, Nitin Madhok notifications@github.com wrote:
> 
> My bad. I didn't think about the recent versions. I was thinking more about supporting users using older versions. Never mind. Thanks for the change!
> 
> —
> Reply to this email directly or view it on GitHub.

nmadhok commented:
That would be great!
